packages/legend-query/src/stores/QueryBuilderExplorerState.ts

```diff
@@ -310,14 +396,36 @@ export const getQueryBuilderPropertyNodeData = (
     mappingNodeData,
   );
   if (propertyNode.type instanceof Class) {
-    propertyNode.childrenIds = propertyNode.type
-      .getAllProperties()
-      .concat(propertyNode.type.getAllDerivedProperties())
-      .map((p) => `${propertyNode.id}.${p.name}`);
+    propertyNode.childrenIds = generateChildrenIds(propertyNode);
   }
   return propertyNode;
 };
 
+export const getQueryBuilderSubTypeNodeData = (
+  subClass: Class,
+  parentNode: QueryBuilderExplorerTreeNodeData,
+): QueryBuilderExplorerTreeSubTypeNodeData => {
+  const subTypeNode = new QueryBuilderExplorerTreeSubTypeNodeData(
+    `${
+      parentNode instanceof QueryBuilderExplorerTreeRootNodeData
+        ? `@${subClass.name}`
```

Comment:
Do me favor, in `MetaModelConst`, create a new token called `export const TYPE_CAST_TOKEN = '@'` and use it, we should avoid using magic strings like this. Also, do me another favor and move `export const VARIABLE_REFERENCE_TOKEN = '$';` to `MetaModelConst` where we define other tokens of the Pure language.